remove rdbms release state wording (#9521)
camunda/camunda-docs: docs/self-managed/quickstart/developer-quickstart/c8run/secondary-storage.md

To use Elasticsearch:

```yaml
camunda:
  data:
    secondary-storage:
      type: elasticsearch
      elasticsearch:
        url: http://localhost:9200/
```

Start Camunda 8 Run with `--config <file>` and point the configuration to your external cluster.

## Switching between storage types and migration notes

- Switching the secondary storage type, for example H2 to Elasticsearch, currently does **not** preserve existing secondary-store data. The system starts with a fresh secondary store.
- To switch storage, update `data.secondary-storage` in `application.yaml` and restart Camunda 8 Run.

Choose **H2** for quick local development and **Elasticsearch** for production-like scenarios where advanced search and analytics are required.

## H2 limitations for Operate

Operate can run against H2, but some user-facing features are limited:

- Operate does not provide complete analytics, advanced search, or long-running query features when backed by H2.
- Performance and scaling behavior when using H2 differs from Elasticsearch.

> - For full Operate feature parity, use an external Elasticsearch instance.

Reviewer: I would propose to remove that

To use Elasticsearch:

```yaml
camunda:
  data:
    secondary-storage:
      type: elasticsearch
      elasticsearch:
        url: http://localhost:9200/
```

Start Camunda 8 Run with `--config <file>` and point the configuration to your external cluster.

## Switching between storage types and migration notes

> - Switching the secondary storage type, for example H2 to Elasticsearch, currently does **not** preserve existing secondary-store data. The system starts with a fresh secondary store.

Reviewer: ```suggestion
- Switching the secondary storage type, for example H2 to Elasticsearch, does **not** preserve existing secondary-store data. The system starts with a fresh secondary store.
```

- To switch storage, update `data.secondary-storage` in `application.yaml` and restart Camunda 8 Run.

> Choose **H2** for quick local development and **Elasticsearch** for production-like scenarios where advanced search and analytics are required.

Reviewer: ```suggestion
Choose **H2** for quick local development and other supported **RDBMS** or **Elasticsearch** for production-like scenarios.
```

For more details on how these layers interact, see [secondary storage architecture](/self-managed/concepts/secondary-storage/index.md).

> Camunda 8 Run uses v2 APIs by default, so no additional configuration is required when switching between secondary storage backends.

Reviewer 2: ```suggestion
Camunda 8 Run uses v2 APIs by default, so you don't need to reconfigure the APIs when switching between secondary storage backends.
```

For more details on how these layers interact, see [secondary storage architecture](/self-managed/concepts/secondary-storage/index.md).

Camunda 8 Run uses v2 APIs by default, so no additional configuration is required when switching between secondary storage backends.

## Known limitations

> - H2 is intended for testing and local development only. For full Operate functionality, use Elasticsearch.

Reviewer: ```suggestion
- H2 is intended for testing and local development only. 
```
